### ISSUE TITLE
Unable to create System.Drawing.Bitmap object with 12712x16910 pixels (73633c43)

### ISSUE THREAD

**Issue by bovender** [bovender/XLToolbox#12]
> This issue was submitted by the add-ins's exception handler.

#### Exception:

````csharp
XLToolbox.Export.ExportException: Unable to create System.Drawing.Bitmap object with 12712x16910 pixels ---> System.ArgumentException: Parameter is not valid.
   at System.Drawing.Bitmap..ctor(Int32 width, Int32 height, PixelFormat format)
   at System.Drawing.Bitmap..ctor(Int32 width, Int32 height)
   at XLToolbox.Export.Exporter.ExportViaFreeImage(Metafile metafile, Preset preset, Double width, Double height, String fileName) in XLToolbox\Export\Exporter.cs:line 277
   --- End of inner exception stack trace ---
   at XLToolbox.Export.Exporter.ExportViaFreeImage(Metafile metafile, Preset preset, Double width, Double height, String fileName) in XLToolbox\Export\Exporter.cs:line 283
   at XLToolbox.Export.Exporter.ExportSelection(Preset preset, Double widthInPoints, Double heightInPoints, String fileName) in XLToolbox\Export\Exporter.cs:line 248
   at XLToolbox.Export.Exporter.ExportSelection(SingleExportSettings settings) in XLToolbox\Export\Exporter.cs:line 77
   at XLToolbox.Export.ViewModels.SingleExportSettingsViewModel.DoExport() in XLToolbox\Export\ViewModels\SingleExportSettingsViewModel.cs:line 201
   at XLToolbox.Export.ViewModels.SingleExportSettingsViewModel.DoConfirmFileName(FileNameMessageContent messageContent) in XLToolbox\Export\ViewModels\SingleExportSettingsViewModel.cs:line 268
   at XLToolbox.Export.ViewModels.SingleExportSettingsViewModel.<DoChooseFileName>b__17(FileNameMessageContent content) in XLToolbox\Export\ViewModels\SingleExportSettingsViewModel.cs:line 243
   at Bovender.Mvvm.Messaging.Message`1.<>c__DisplayClass2.<Send>b__0()
   at Bovender.Mvvm.Actions.FileDialogActionBase.Invoke(Object parameter)
   at System.Windows.Interactivity.TriggerBase.InvokeActions(Object parameter)
   at System.Windows.Interactivity.EventTriggerBase.OnEvent(EventArgs eventArgs)
   at System.Windows.Interactivity.EventTriggerBase.OnEventImpl(Object sender, EventArgs eventArgs)
   at Bovender.Mvvm.Messaging.Message`1.Send(T messageContent, Action`1 respond)
   at XLToolbox.Export.ViewModels.SingleExportSettingsViewModel.DoChooseFileName() in XLToolbox\Export\ViewModels\SingleExportSettingsViewModel.cs:line 238
   at XLToolbox.Export.ViewModels.SingleExportSettingsViewModel.<get_ChooseFileNameCommand>b__6(Object param) in XLToolbox\Export\ViewModels\SingleExportSettingsViewModel.cs:line 139
   at Bovender.Mvvm.DelegatingCommand.Execute(Object parameter)
````

Message: Unable to create System.Drawing.Bitmap object with 12712x16910 pixels
#### Inner exception:

````csharp
System.ArgumentException: Parameter is not valid.
   at System.Drawing.Bitmap..ctor(Int32 width, Int32 height, PixelFormat format)
   at System.Drawing.Bitmap..ctor(Int32 width, Int32 height)
   at XLToolbox.Export.Exporter.ExportViaFreeImage(Metafile metafile, Preset preset, Double width, Double height, String fileName) in XLToolbox\Export\Exporter.cs:line 277
````

Inner exception message: Parameter is not valid.

**Comment by bovender** [comment]
The exception message indicates that the error occurred on attempting to create an image with 12712 x 16910 pixels, i.e. about 215 Megapixels. At 300 dpi, the image size would be 42 x 56 inches (107 x 142 cm, which is considerably larger than A0.

The add-in currently cannot handle such a large bitmap, as it would require obtaining a chunk of memory that is ~800 megabytes large. It might be possible with tiling. But on the other hand, I personally wonder if such an extremely high-resolution image with > 200 megapixels was really intended.

**Comment by bovender** [comment]
See #16.

Fixed in 3510106a02046dc13ee50b526cd785dac417c408. Will be in v7.0.0-beta.7.
